scripts/cxg_admin.py

```diff
@@ -558,6 +558,90 @@ def add_trailing_slash_to_explorer_urls(ctx):
+
+    AUTH0_CLIENT_ID: Retrieved from Auth0 console in the application section under Client ID. The application must be
+    authorized to access to the Auth0 Management API with the following permissions read:users read:user_idp_tokens.
+
+    AUTH0_CLIENT_SECRET: Retrieved from Auth0 console in the application section under Client Secret.
+    """
+
+    auth0_api = auth0_apis[ctx.obj["deployment"]]
+
+    def get_auth0_access_token():
+        logger.info("Retrieving access token from Auth0.")
+        payload = json.dumps(
+            {
+                "client_id": auth0_client_id,
+                "client_secret": auth0_client_secret,
+                "audience": f"{auth0_api}/api/v2/",
+                "grant_type": "client_credentials",
+            }
+        )
+        headers = {"content-type": "application/json"}
+        response = requests.post(f"{auth0_api}/oauth/token", headers=headers, data=payload)
+        access_token = response.json()["access_token"]
+        return access_token
+
+    def get_owners_from_database():
+        logger.info("Gathering owners from collections.")
+        owners = []
+        with db_session_manager() as session:
+            for collection in Collection.list(session):
+                owner = collection.owner
+                if owner:
+                    owners.append(collection.owner)
+        unique_owners = set(owners)
+        return unique_owners
+
+    def get_owner_info_from_auth0(owner, access_token):
+        logger.info(f"Retrieving curator info for owner:{owner} from Auth0.")
+        url_manager = furl(
+            f"{auth0_api}/api/v2/users/{owner}",
+            query_params=dict(fields=",".join(["given_name", "family_name"]), include_fields="true"),
+        )
+        response = requests.get(url_manager.url, headers={"Authorization": f"Bearer {access_token}"})
+        response.raise_for_status()
+        if response.headers["X-RateLimit-Remaining"] == 0:
+            wait = time.time() - response.headers["X-RateLimit-Reset"]
+            time.sleep(wait)
+        body = response.json()
+        return f"{body.get('given_name', '')} {body.get('family_name', '')}".strip()
+
+    def update_database_curator_name(owner_id, owner_name):
+        logger.info(f"Updating collections owned by {owner_id} with curator_name:{owner_name}.")
+        with db_session_manager() as session:
+            collections = session.query(DbCollection).filter(DbCollection.owner == owner_id).all()
+            for collection in collections:
+                collection.curator_name = owner_name
+                collection.data_submission_policy_version = "2.0"
+
+    access_token = get_auth0_access_token()
+    unique_owners = get_owners_from_database()
+    owners = dict()
+    for owner in unique_owners:
+        try:
+            owner_name = get_owner_info_from_auth0(owner, access_token)
+            if owner_name:
+                owners[owner] = owner_name
```

Comment:
Should we do a check here that owner_name isn't an empty string?

Reply:
in python an empty string == False

Reply:
Sorry, should have elaborated, do we want to notify the user when there is an owner with no name attached to their account?

Reply:
We can run this to populate all of the curators and then run another query to see what collections have no curator.